module/cmd/peggy/cmd/gentx.go

```diff
@@ -281,246 +277,3 @@ func writeSignedGenTx(clientCtx client.Context, outputDocument string, tx sdk.Tx
-
-	// if there are no gen txs to be processed, return the default empty state
-	if len(appGenTxs) == 0 {
-		return appState, errors.New("there must be at least one genesis tx")
-	}
-
-	// create the app state
-	appGenesisState, err := types.GenesisStateFromGenDoc(genDoc)
-	if err != nil {
-		return appState, err
-	}
-
-	appGenesisState, err = genutil.SetGenTxsInAppGenesisState(cdc, txEncodingConfig.TxJSONEncoder(), appGenesisState, appGenTxs)
-	if err != nil {
-		return appState, err
-	}
-
-	appState, err = json.MarshalIndent(appGenesisState, "", "  ")
-	if err != nil {
-		return appState, err
-	}
-
-	genDoc.AppState = appState
-	err = genutil.ExportGenesisFile(&genDoc, config.GenesisFile())
-
-	return appState, err
-}
-
-// CollectTxs processes and validates application's genesis Txs and returns
-// the list of appGenTxs, and persistent peers required to generate genesis.json.
-func CollectTxs(cdc codec.JSONMarshaler, txJSONDecoder sdk.TxDecoder, moniker, genTxsDir string,
-	genDoc tmtypes.GenesisDoc, genBalIterator types.GenesisBalancesIterator,
-) (appGenTxs []sdk.Tx, persistentPeers string, err error) {
-	// prepare a map of all balances in genesis state to then validate
-	// against the validators addresses
-	var appState map[string]json.RawMessage
-	if err := json.Unmarshal(genDoc.AppState, &appState); err != nil {
-		return appGenTxs, persistentPeers, err
-	}
-
-	var fos []os.FileInfo
-	fos, err = ioutil.ReadDir(genTxsDir)
-	if err != nil {
-		return appGenTxs, persistentPeers, err
-	}
-
-	balancesMap := make(map[string]bankexported.GenesisBalance)
-
-	genBalIterator.IterateGenesisBalances(
-		cdc, appState,
-		func(balance bankexported.GenesisBalance) (stop bool) {
-			balancesMap[balance.GetAddress().String()] = balance
-			return false
-		},
-	)
-
-	// addresses and IPs (and port) validator server info
-	var addressesIPs []string
-
-	for _, fo := range fos {
-		if fo.IsDir() {
-			continue
-		}
-		if !strings.HasSuffix(fo.Name(), ".json") {
-			continue
-		}
-
-		// get the genTx
-		jsonRawTx, err := ioutil.ReadFile(filepath.Join(genTxsDir, fo.Name()))
-		if err != nil {
-			return appGenTxs, persistentPeers, err
-		}
-
-		var genTx sdk.Tx
-		if genTx, err = txJSONDecoder(jsonRawTx); err != nil {
-			return appGenTxs, persistentPeers, err
-		}
-
-		appGenTxs = append(appGenTxs, genTx)
-
-		// the memo flag is used to store
-		// the ip and node-id, for example this may be:
-		// "528fd3df22b31f4969b05652bfe8f0fe921321d5@192.168.2.37:26656"
-
-		memoTx, ok := genTx.(sdk.TxWithMemo)
-		if !ok {
-			return appGenTxs, persistentPeers, fmt.Errorf("expected TxWithMemo, got %T", genTx)
-		}
-		nodeAddrIP := memoTx.GetMemo()
-		if len(nodeAddrIP) == 0 {
-			return appGenTxs, persistentPeers, fmt.Errorf("failed to find node's address and IP in %s", fo.Name())
-		}
-
-		// genesis transactions must be single-message
-		msgs := genTx.GetMsgs()
-		// if len(msgs) != 1 {
```

Comment:
So the one issue here is that collect gentxs currently errors when `len(msgs) != 1`. Also I think we may want to add some other custom validation for gentxs that will help us start chains so I think we should leave this.

Reply:
cc @fedekunze